modules/gdscript/gdscript_byte_codegen.cpp

```diff
@@ -585,8 +585,61 @@ void GDScriptByteCodeGenerator::write_unary_operator(const Address &p_target, Va
 }
 
 void GDScriptByteCodeGenerator::write_binary_operator(const Address &p_target, Variant::Operator p_operator, const Address &p_left_operand, const Address &p_right_operand) {
-	// Avoid validated evaluator for modulo and division when operands are int, since there's no check for division by zero.
-	if (HAS_BUILTIN_TYPE(p_left_operand) && HAS_BUILTIN_TYPE(p_right_operand) && ((p_operator != Variant::OP_DIVIDE && p_operator != Variant::OP_MODULE) || p_left_operand.type.builtin_type != Variant::INT || p_right_operand.type.builtin_type != Variant::INT)) {
+	// Avoid validated evaluator for modulo and division when operands are int or integer vector, since there's no check for division by zero.
+	bool valid = HAS_BUILTIN_TYPE(p_left_operand) && HAS_BUILTIN_TYPE(p_right_operand);
+	if (valid && (p_operator == Variant::OP_DIVIDE || p_operator == Variant::OP_MODULE)) {
+		switch (p_left_operand.type.builtin_type) {
+			case Variant::INT:
+				switch (p_right_operand.type.builtin_type) {
+					case Variant::INT:
+						valid = false;
+						break;
+					default:
+						break;
+				}
+				break;
+			case Variant::VECTOR2I:
+				switch (p_right_operand.type.builtin_type) {
+					case Variant::INT:
+						valid = false;
+						break;
+					case Variant::VECTOR2I:
+						valid = false;
+						break;
```

Comment:
This can be simplified to:

```suggestion
					case Variant::INT:
					case Variant::VECTOR2I:
						valid = false;
						break;
```

Same for the other cases below.